validations.py

```diff
@@ -0,0 +1,180 @@
+            global_filter = ignore_list['global_filter']
+
+            if 'error_patterns' in global_filter and type(global_filter['error_patterns']) is list:
+                global_filters = global_filter['error_patterns']
+
+    return global_filters
+
+
+def get_file_filter(ignore_list, file_path):
+    filters = []
+
+    if type(ignore_list) is dict:
+        if 'file_filter' in ignore_list and ignore_list['file_filter'] is not None:
+            for file_filter in ignore_list['file_filter']:
+                if re.search(file_filter['file_pattern'], file_path):
+                    if 'error_patterns' in file_filter and type(file_filter['error_patterns']) is list:
+                        filters = filters + file_filter['error_patterns']
+
+    return filters
+
+
+def filter_errors(ignore_list, errors_per_file):
+    global_filters = get_global_filter(ignore_list)
+
+    filtered_errors_per_file = {}
+
+    for file_name in list(errors_per_file.keys()):
+        file_filters = get_file_filter(ignore_list, file_name)
+
+        for error in errors_per_file[file_name]:
+            if not any_filters_match(global_filters, error) and not any_filters_match(file_filters, error):
+                filtered_errors_per_file.setdefault(file_name, []).append(error)
+
+    return filtered_errors_per_file
+
+
+def validate_resources(version, ignore_list, files):
+    java_commands = ['java', '-Xmx3G', '-Xms2G', '-jar', 'validator_cli.jar']
+
+    for file_name in files:
+        java_commands.append(file_name)
+
+    java_commands.append('-version')
+    java_commands.append(version)
+
+    java_call = subprocess.run(java_commands, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output = java_call.stdout.decode('utf-8')
+    error_output = java_call.stderr.decode('utf-8')
+    if java_call.returncode != 0:
+        if output == '':
+            print('Validation of \'{}\' failed, but no output was generated.'.format(file_path))
+        elif error_output != '':
+            print(error_output)
+            print('There was an issue processing the request.')
+            exit(1)
+    output_lines = output.split('\n')
+
+    file_name = "loading_output"
+    errors_per_file = {}
+
+    for line in output_lines:
+        if line.startswith('-- '):
+            file_search = re.search('-- ([^\\s]*) -*', line)
+
+            if file_search:
+                file_name = file_search.group(1)
+                errors_per_file[file_name] = []
+
+        stripped_line = line.strip()
+
+        if stripped_line.lower().startswith('error @'):
+            errors_per_file[file_name].append(stripped_line)
+
+    return filter_errors(ignore_list, errors_per_file)
+
+
+def validate_resource_dir(run_config, ignore_list, recently_changed):
+    print('Checking directory {}'.format(run_config.target_dir))
+    files = get_fhir_resource_files(run_config.target_dir, recently_changed)
```

Comment:
The idea was to keep the script simple to use, as I was trying to avoid having a fairly elaborate readme about how you have to run a `git diff` and pipe the results to the script.  The fact that there is a seemingly redundant `git diff` done in the pipeline was for the sake of pipeline efficiency, and technically serves a different purpose than the one in the script, even though it's the same command.